gtp2ogs.js

```diff
@@ -958,44 +936,94 @@ class Game {
             }
-            if (move.resign) {
-                this.log("Resigning");
-                this.socket.emit('game/resign', this.auth({
-                    'game_id': this.state.game_id
-                }));
-            }
-            else {
-                if (DEBUG) this.log("Playing " + move.text, move);
-                else       this.log("Playing " + move.text);
-                this.socket.emit('game/move', this.auth({
-                    'game_id': this.state.game_id,
-                    'move': encodeMove(move)
-                }));
-                //this.sendChat("Test chat message, my move #" + move_number + " is: " + move.text, move_number, "malkovich");
-        if( argv.greeting && !this.greeted && this.state.moves.length < (2 + this.state.handicap) ){
-                  this.sendChat( GREETING, "discussion");
-                  this.greeted = true;
+	};
+
+        let failed = false;
+        let botError = (e) => {
+            if (failed)  return;
+	    
+            failed = true;
+	    doneProcessing();
+            if (this.bot) this.bot.kill();
+            this.bot = null;
+	    if (eb) eb(e);
         }
-            }
-            if (!PERSIST && this.bot != null) {
+
+        if (!this.bot)  this.startBot(botError);
+
+	if (DEBUG) this.bot.log("Generating move for game", this.game_id);
+	this.log("genmove " + (this.my_color == "black" ? "b" : "w"));
+
+        this.bot.getMoves(cmd, this.state, get_handicap_stones, (moves) => {
+	    doneProcessing();
+	    cb(moves)
+	    
+	    if (!PERSIST && this.bot != null) {
                 this.bot.kill();
                 this.bot = null;
-            }
-        }, passAndRestart);
+	    }
+        }, botError);
+    } /* }}} */
+
+    // Send move to server.
+    // 
+    uploadMove(move) { /* {{{ */	    
+	if (move.resign) {
+            this.log("Resigning");
+            this.socket.emit('game/resign', this.auth({
+		'game_id': this.state.game_id
+            }));
+	    return;
+	}
+	
+        if (DEBUG) this.log("Playing " + move.text, move);
+	else       this.log("Playing " + move.text);
+        this.socket.emit('game/move', this.auth({
+	    'game_id': this.state.game_id,
+	    'move': encodeMove(move)
+        }));
+        //this.sendChat("Test chat message, my move #" + move_number + " is: " + move.text, move_number, "malkovich");
+	if( argv.greeting && !this.greeted && this.state.moves.length < (2 + this.state.handicap) ){
+	    this.sendChat( GREETING, "discussion");
+	    this.greeted = true;
+	}
     } /* }}} */
+
+    // Get move from bot and upload to server.
+    //
+    makeMove(move_number) { /* {{{ */
+        if (DEBUG && this.state) { this.log("makeMove", move_number, "is", this.state.moves.length, "!=", move_number, "?"); }
+        if (!this.state || this.state.moves.length != move_number)
+            return;
+        if (this.state.phase != 'play')
+            return;
+
+	let sendMove = (moves) => {  this.uploadMove(moves[0]);  };
```

Comment:
Should this be called sendFirstMove as it silently ignores the rest of the moves?

Reply:
Or sendTheMove(), anything is good with me.